app/models/manageiq/providers/vmware/infra_manager/provision/placement.rb

```diff
@@ -15,25 +15,39 @@ def placement
 
   def manual_placement
     _log.info("Manual placement...")
-    return selected_placement_obj(:placement_host_name, Host),
-           selected_placement_obj(:placement_ds_name, Storage)
+
+    host      = selected_placement_obj(:placement_host_name, Host)
+    cluster   = selected_placement_obj(:placement_cluster_name, EmsCluster)
+    datastore = selected_placement_obj(:placement_ds_name, Storage)
+
+    raise MiqException::MiqProvisionError, "Destination placement_ds_name not provided" if datastore.nil?
+    raise MiqException::MiqProvisionError, "Destination placement_host_name and placement_cluster_name not provided" if host.nil? && cluster.nil?
```

Comment:
The `selected_placement_obj` method used by both the `manual_placement` and `automatic_placement` methods use to raise an error if the object was not defined.  Now the `manual_placement` method is checking here and the `automatic_placement` will not catch the error until `prepare_for_clone_task`.

Can we  consolidate these checks in the `placement` method above?  I would move the new `raise` logic in `prepare_for_clone_task` and do that in the `placement` method as well.  It makes more sense to do all our placement checking in that one place.